tianshou/data/utils.py

```diff
@@ -36,14 +40,16 @@ def to_torch(x: Union[torch.Tensor, dict, Batch, np.ndarray],
         x.to_torch(dtype, device)
     elif isinstance(x, (np.number, np.bool_, Number)):
         x = to_torch(np.asanyarray(x), dtype, device)
-    elif isinstance(x, list) and len(x) > 0 and \
-            all(isinstance(e, (np.number, np.bool_, Number)) for e in x):
-        x = to_torch(np.asanyarray(x), dtype, device)
-    elif isinstance(x, np.ndarray) and \
-            isinstance(x.item(0), (np.number, np.bool_, Number)):
-        x = torch.from_numpy(x).to(device)
-        if dtype is not None:
-            x = x.type(dtype)
+    elif isinstance(x, (list, tuple)):
+        x = [to_torch(x_i, dtype, device) for x_i in x]
+    else:  # fallback
+        x = np.asanyarray(x)
+        if isinstance(x.item(0), (np.number, np.bool_, Number)):
```

Comment:
Docstring:
a.item(*args)
Copy an element of an array to a standard Python scalar and return it.

It seems x.item(0) will not be instance of np.number or np.bool_